File: app/XDM/XDM.Core/StreamingVideoDisplayInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using XDM.Core.Util;

namespace XDM.Core
{
    public struct StreamingVideoDisplayInfo
    {
        public string Quality { get; set; }
        public long Size { get; set; }
        public long Duration { get; set; }
        public DateTime CreationTime { get; set; }
        public string TabUrl { get; set; }

        public string DescriptionText
        {
            get
            {
                var sb = new StringBuilder();
                //if (Size > 0)
                //{
                //    sb.Append(FormattingHelper.FormatSize(Size));
                //}
                //AddSpece(sb);
                if (!string.IsNullOrEmpty(Quality))
                {
                    sb.Append(Quality);
                }
                AddSpece(sb);
                if (Duration > 0)
                {
                    sb.Append(FormattingHelper.ToHMS(Duration));
                }
                return sb.ToString();
            }
        }

        private void AddSpece(StringBuilder sb)
        {
            if (sb.Length > 0)
            {
                sb.Append(" ");
            }
        }
    }
}

File: xdm-app/src/main/java/xdm/app/models/StreamingVideoDisplayInfo.kt
package xdm.app.models

import xdm.core.util.FormatUtilities
import java.time.LocalDateTime

data class StreamingVideoDisplayInfo(
    var quality: String? = null,
    var size: Long? = null,
    var duration: Long? = null,
    var dateTime: LocalDateTime = LocalDateTime.now(),
    var tabUrl: String? = null,
    var tabId: String? = null,
) {
    val descriptionText: String
        get() {
            return StringBuilder().apply {
                quality?.let { append(it) }
                duration?.let {
                    if (it > 0) {
                        quality?.let { append(" ") }
                        append(FormatUtilities.hms(it.toInt()))
                    }
                }
            }.toString()
        }
}